src/hif/hif_load.jl

```diff
@@ -0,0 +1,134 @@
+using JSON3
+
+"""
+    hg_load(
+        io::IO,
+        format::HIF_Format;
+        T::Type{U} = Bool,
+        D::Type{<:AbstractDict{Int, U}} = Dict{Int,U},
+    ) where {U <: Real}
+
+Loads a hypergraph from a stream `io` from `HIF` format.
+More info: https://github.com/pszufe/HIF-standard
+
+**Arguments**
+
+* `T` : type of weight values stored in the hypergraph's adjacency matrix
+* `D` : dictionary for storing values the default is `Dict{Int, T}`
+"""
+function hg_load(
+    io::IO,
+    format::HIF_Format;
+    T::Type{U} = Bool,
+    D::Type{<:AbstractDict{Int,U}} = Dict{Int,T},
+) where {U<:Real}
+    _ = format
+
+    data = JSON3.read(read(io, String))
+
+    if !haskey(data, "incidences")
+        throw(ArgumentError("Invalid JSON schema: missing required key 'incidences'"))
+    end
+
+    nodes, edges = get_nodes_and_edges(data)
+
+    h = init_hypergraph(data, length(nodes), length(edges), T, D)
+
+    add_weights_from_incidences!(h, data.incidences, nodes, edges)
+
+    h
+end
+
+
+function init_hypergraph(
+    data::JSON3.Object,
+    n::Int64,
+    k::Int64,
+    T::Type{U},
+    D::Type{<:AbstractDict{Int,U}},
+) where {U<:Real}
+    node_metadata = Vector{Union{JSON3.Object, Nothing}}()
```

Comment:
You do not want the user suddenly see some `JSON3.Object` just cast it to a `String`.
Additionally, the code could check if this is some more complex data structure and than make it into a Dict.
Perhaps what could help would be using second parameter for reading `JSON3.read("test.json", Dict{String, Any})`